src/abstractdataframe/abstractdataframe.jl

```diff
@@ -654,8 +654,7 @@ julia> dropmissing(df, [:x, :y])
 ```
 
 """
-function dropmissing(df::AbstractDataFrame,
-                     cols::Union{Integer, Symbol, AbstractVector, Regex}=1:size(df, 2);
+function dropmissing(df::AbstractDataFrame, cols=1:size(df, 2);
```

Comment:
Why not keep the types? That makes things clearer both for developers and for users (via early errors).

Reply:
I can reintroduce them. The problem was that you still have to add `Colon` to the list and it seemed overly long (this correct handling of `Colon` is one of the small things I clean in this PR).

Reply:
and also `Not` has to be added

Reply:
Yes, it's verbose, but that's not the end of the world. We can create an alias if needed (a much more useful one than the current `ColumnIndex`).

Reply:
I was thinking about an alias - I will create it.

Reply:
Actually it is not that easy - in different places we split the `Union{AbstractVector, Regex, Not, Colon}` in different ways to handle the specific cases fast. Maybe I will leave it as is explicitly.

Reply:
Though if there's a more specific `Union`, it should take precedence in dispatch.

Reply:
Yes, but then users might get confused. I propose to leave it explicit.

Reply:
OK. But better keep the `Union` in the signature. Same in other functions.

Reply:
fixed. I have also patched some more docstrings.